examples/ecs/observers.rs

```diff
@@ -0,0 +1,206 @@
+//! Demonstrates how to observe life-cycle triggers as well as define custom ones.
+
+use bevy::{
+    ecs::observer::Observer,
```

Comment:
`Observer` should probably be in the prelude, assuming it should be commonly used as a component

Reply:
Agreed!